src/main/java/edu/greenblitz/tobyDetermined/subsystems/DriveTrain/IO/GyroIOs/Gyro.java

```diff
@@ -0,0 +1,53 @@
+package edu.greenblitz.tobyDetermined.subsystems.DriveTrain.IO.GyroIOs;
+
+import edu.greenblitz.tobyDetermined.RobotMap;
+
+public class Gyro {
```

Comment:
Okay, you tried to use this class to both use a generic gyro and decide which type of gyro to use, this class has 2 purposes (which is 1 too much).
Search online about the 'Factory' design pattern.